### PR TITLE
Add unsupervised keyphrase extraction

### PR DESCRIPTION
**Pull request by bartdegoede**
Follow up on https://github.com/textpipe/textpipe/pull/73

It adds an `extract_keyphrases` method rather than a property, because there's three different algorithms that we can use. This way, it's up to the user what to pick. The [`sgrank` algorithm](https://github.com/chartbeat-labs/textacy/blob/master/textacy/keyterms.py#L29-L30) provides some additional arguments (selecting which ngrams to consider, for example), so I've added a `**kwargs` for users to pass these arguments on.

### CODE REVIEW

**Comment by anneschuth** [comment]
Nice! Additionally I'd be in favour of adding a property that selects the best (up to you, or us).

File: textpipe/doc.py [hunk 1]
```diff
@@ -314,3 +315,39 @@ def sentiment(self):
             return sentiment_it(self.clean)
 
         raise TextpipeMissingModelException(f'No sentiment model for {self.language}')
+
+    def extract_keyphrases(self, ranker='textrank', n_terms=10, **kwargs):
```
Comment (anneschuth) [inline textpipe/doc.py]: Can you add a LRU cache to this method?

**Comment by anneschuth** [comment]
Don't forget to bump the version to 0.6.0

**Comment by bartdegoede** [comment]
In terms of choosing the "best" ranker, the reason why `textrank` is the default is that it's easiest to understand (I'm having a hard time deciding what the "best" would be in which usecase without some extensive tests 😁). Any suggestions?

**Comment by anneschuth** [comment]
Cool, one last thing: can you add a matching operation (in operation.py)?

**Comment by bartdegoede** [comment]
FYI, I rebased my changes of your latest one in `master` :-)

File: VERSION [hunk 2]
```diff
@@ -1 +1 @@
-0.5.2
+0.5.3
```
Comment (anneschuth) [inline VERSION]: could be 0.6.0 bump

Comment (anneschuth) [reply VERSION]: or should, I think

Comment (bartdegoede) [reply VERSION]: Change is that major? ;-)

Comment (anneschuth) [reply VERSION]: no, minor :) major would increment the first digit.

minor is for added functionality
the last digit is for patches / bug fixes

**Comment by msappelli** [comment]
Does textrank only extract 1-grams or also actual phrases? I only see a multi word term in the sgrank example. If the function is called 'extract_keyphrases' I expect more/only multi-word terms. otherwise use something like extract_keyterms as a name to be clear about the functionality.

**Comment by bartdegoede** [comment]
@msappelli [It doesn't](https://github.com/chartbeat-labs/textacy/blob/master/textacy/keyterms.py#L200). Textacy hardcodes the parameter for joining terms for `textrank` to `False`. The other two do, and I went for `phrases` because I felt that would communicate better to users what would be returned (1+ grams, depending on the ranker).

That's totally subjective though, so if you (the maintainers) feel `keyterms` is more appropriate, I'm more than happy to change that :-)

**Comment by msappelli** [comment]
@bartdegoede I don't think it's subjective. I see that in the IR community keyterm/ keyword / keyphrase is often used interchangeably. But in NLP/linguistics there is a difference, where 'keyterm' is typically 1-2 or 3 gram (so the interpretation of 'term' is not necessarily a 1-gram, but ), whereas a (key) phrase is a group of words (>1) that have a certain grammatical or semantic function.

Additionally keyterms would be consistent with how textacy calls it, and since we are using their function underlying I would keep it consistent.

**Comment by bartdegoede** [comment]
Awesome, makes total sense :-) Changing it now.

**Comment by msappelli** [comment]
ok

**Pull request merged**